src/chain-data/context.tsx

```diff
@@ -35,4 +37,37 @@ const ChainDataContextProvider = (props: { children: ReactNode }) => {
 
 export default ChainDataContextProvider;
 
-export const useChainData = () => useContext(ChainDataContext);
+export const useChainData = () => {
+  const data = useContext(ChainDataContext);
+  const { chain } = useNetwork();
+  const provider = useProvider({ chainId: chain?.id });
+  const { data: signer } = useSigner({ chainId: chain?.id });
+
+  const { isConnected, address } = useAccount();
+
+  // Note: The signer is briefly undefined after connecting or after switching networks.
+  if (isConnected && address && signer) {
+    const networkName = updateNetworkName(chain?.network || '');
+    return {
+      ...data,
+      isConnected: true,
+      provider,
+      signer,
```

Comment:
Since we are using wagmi, it doesn't make sense to have data like `signer`, `provider`, `userAccount` in our context. I expect it is a larger (and tedious) refactor to remove it - so better done in separate issue.

Reply:
Nevermind. I see that you did something even better - modified the `useChainData` hook to return combination of our context and wagmi context. Very nice 🥂